openhtf/output/web_gui/src/README.md

```diff
@@ -0,0 +1,134 @@
+# How to write a Plugin! 
+
+It is a truth universally acknowledged that an application with infinite 
+use cases must be in want of a plugin interface. Here's how to write one!
+
+## Contents
+
+A basic, bare-bones plugin package structure looks like this: 
+```
+/example_plugin
+	|
+	-- setup.py
+	|
+	-- /example_plugin
+			|
+			-- __init__.py
+			|
+			-- /plugin
+				|
+				-- main.ts
+				|
+				-- example_plugin.html
+				|
+				-- example_service.ts
+```
+
+A plugin with this structure would be able to show up in the frontend and 
+make calls to the backend Tornado server to request information.
+
+### Backend Structure
+
+Everything adding to the backend goes in the __init__.py module! It requires 
+three main parts: 
+
+#### Handler Variable
+
+This tuple is what OpenHTF will use to configure and add your tornado handler. 
+A simple example: 
+```
+handler=([r'/example/', ExampleHandler],
+		 ['host','port'])
+```
+The tuple consists of two parts: 
+- A list containing the regex for the URL you need served, and the name 
+of the handler to serve it to 
+- A list of backend dependencies needed in your handler, an empty list 
+if none needed. Currently, you can get references to the:
+	- station store
+	- host
+	- port
+	- path
+
+#### Configs Variable 
+
+This dictionary specifies your configurations for the frontend portions of 
+the plugin. Make sure these variables match the class and tag you use in your 
+main.ts module; this component will populate the body of your plugin tab.
+Another example: 
+```
+configs={
+	'class':'ExamplePlugin',
+	'tag':'example-plugin', 
+	'label':'example', 
+	'icon':'pets'}
+```
+- label: The label for the tab for your plugin
+- icon: Optional, but refers to the icon from Material Icons that you want
+on your tab. (replace any spaces with underscores)
+
+#### Tornado Handler
+
+If you need to pull information from the backend, then you need to create a 
```

Comment:
This is the handler mentioned in the variable above? If it's optional, I'd mention that above and combine this section with that one since it's confusing to separate them.